File: src/carlot.ui/src/types/CarDto.ts
import type { EquipmentDto } from "./EquipmentDto";

export interface CarDto {
  id: string;
  vin: string;
  make: string;
  model: string;
  year: number;
  fuelType: FuelType;
  transmission: TransmissionType;
  additionalFuelType: AdditionalFuelType;
  powerHp: number;
  engineDisplacement?: number;
  turbocharged: boolean;
  body: string;
  registrationPlate: string;
  driveType: DriveType;
  mileageKm: number;
  location: string;
  version: number;
  createdAtUtc: string;
  updatedAtUtc?: string;
  status: CarStatus;
  equipment: ReadonlyArray<EquipmentDto>;
}

export enum FuelType {
  Petrol,
  Diesel,
  Electric,
  Hybrid
}

export enum AdditionalFuelType {
  None,
  LPG,
  CNG
}

export enum TransmissionType {
  Manual,
  Automatic
}

export enum DriveType {
  FWD,
  RWD,
  AWD,
  FourWD
}

export enum CarStatus {
  Active,
  Inactive,
  Sold
}

File: src/CarLot.Listings/CarLot.Listings.Application/DTOs/CarDto.cs
using CarLot.Core;
using DriveType = CarLot.Core.DriveType;

namespace CarLot.Listings.Application.DTOs;

public record CarDto(
    Guid Id,
    string Vin,
    string Make,
    string Model,
    int Year,
    FuelType FuelType,
    Transmission Transmission,
    AdditionalFuelType AdditionalFuelType,
    int PowerHp,
    float? EngineDisplacement,
    bool Turbocharged,
    string Body,
    string RegistrationPlate,
    DriveType DriveType,
    int MileageKm,
    string Location,
    int Version,
    DateTime CreatedAtUtc,
    DateTime? UpdatedAtUtc,
    CarStatus Status,
    IReadOnlyCollection<EquipmentDto> Equipment
);